src/eventSubscription/createWizard/ContainerRegistryListStep.ts

```diff
@@ -0,0 +1,38 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import { Registry } from 'azure-arm-containerregistry/lib/models';
+import { AzureWizardPromptStep, IAzureQuickPickItem, IAzureQuickPickOptions, IAzureUserInput, ISubscriptionWizardContext } from 'vscode-azureextensionui';
+import { localize } from '../../utils/localize';
+
+export interface IContainerRegistryWizardContext extends ISubscriptionWizardContext {
+    registry?: Registry;
+}
+
+export class ContainerRegistryListStep<T extends IContainerRegistryWizardContext> extends AzureWizardPromptStep<T> {
+    public async prompt(wizardContext: T, ui: IAzureUserInput): Promise<T> {
+        if (!wizardContext.registry) {
+            const quickPickOptions: IAzureQuickPickOptions = { placeHolder: localize('listPlaceHolder', 'Select a registry'), id: `ContainerRegistryListStep/${wizardContext.subscriptionId}` };
+            wizardContext.registry = (await ui.showQuickPick(this.getQuickPicks(wizardContext), quickPickOptions)).data;
+        }
+
+        return wizardContext;
+    }
+
+    private async getQuickPicks(wizardContext: T): Promise<IAzureQuickPickItem<Registry>[]> {
+        const client: ContainerRegistryManagementClient = new ContainerRegistryManagementClient(wizardContext.credentials, wizardContext.subscriptionId);
+        const registries: Registry[] = await client.registries.list();
```

Comment:
Given this line, I'm slightly cautious about the non-null assertion a few lines before. Pertaining mainly to [this issue in cosmosdb](https://github.com/Microsoft/vscode-cosmosdb/issues/628), which can be fixed with a `try-catch` similar to [this PR you pointed out](https://github.com/Microsoft/vscode-azureappservice/pull/373/files#diff-4825a11def7265b6f47fe8e35948f738R37)

Your thoughts?

Reply:
Do you mean the non-null assertion a few lines _after_? The issue you referenced in Cosmos DB is for the tree view, which is much more complicated than the simple quick pick in this case. I don't think a try-catch is necessary here - worst case an invalid registry doesn't have a label, but it shouldn't throw an exception and block other registries from being listed.